### PR TITLE
#4 - subroutines coupler_full_chksum and coupler_stocks

### PR DESCRIPTION
**Pull request by mlee03**
In this PR,

The back-to-back calls to `atmos_ice_land_chksum` and  `ocean_chksum`  are combined and replaced with a new subroutine called **coupler_atmos_ice_land_ocean_chksum**.   This conservative PR did not attempt to incorporate subroutines `coupler_chksum` to **coupler_atmos_ice_land_ocean_chksum**.  Doing so would have led to CHECKSUM-related changes in stdout.

In addition,  the subroutines **coupler_flux_init_finish_stocks** and **coupler_flux_check_stocks** have been created and are used to replace stock-related sections of coupler_main.  In coupler_flux_init_finish_stocks, either `flux_init_stocks` or the `final _flux_check_stocks` will be called depending on the  values `init_stocks` and `finish_stocks` optional arguments. 

AM4, AM5, SPEAR, OM4_extra, and OMIP4_CORE2 regression tests pass.  The CHECKSUMS in stdout were not checked for ensemble regression tests.

### CODE REVIEW

File: full/full_coupler_mod.F90 [hunk 1]
```diff
@@ -1493,15 +1499,208 @@ end subroutine slow_ice_chksum
...
+    type(FmsTime_type),    intent(in) :: Time    !< current Time
+    type(atmos_data_type), intent(inout) :: Atm  !< Atm
+    type(land_data_type),  intent(inout) :: Land !< Land
+    type(ice_data_type),   intent(inout) :: Ice  !< Ice
+    type(ocean_state_type), pointer, intent(inout) :: Ocean_state    !< Ocean_state
+    type(coupler_clock_type), intent(inout)        :: coupler_clocks !< coupler_clocks
+    logical, optional, intent(in) :: init_stocks, finish_stocks  !< control flags to either call flux_init_stocks or
+                                                                 !! the final flux_check_stocks
+
+    logical :: init, finish
```
Comment (thomas-robinson) [inline full/full_coupler_mod.F90:1662]: Add documentation for these variables

**Review by thomas-robinson — DISMISSED** [review]

File: full/full_coupler_mod.F90 (hunk 1 above)
Comment (mlee03) [reply full/full_coupler_mod.F90]: @thomas-robinson

File: full/coupler_main.F90 [hunk 2]
```diff
@@ -439,27 +439,19 @@ program coupler_main
...
-                  Land_ice_atmos_boundary, Atmos_ice_boundary, Atmos_land_boundary)
-      endif
-      if (Ocean%is_ocean_pe) then
-        call fms_mpp_set_current_pelist(Ocean%pelist)
-        call ocean_chksum('MAIN_LOOP-', nc, Ocean, Ice_ocean_boundary)
-      endif
-      call fms_mpp_set_current_pelist()
-    endif
+    if (do_chksum) then      
+      call coupler_chksum('top_of_coupled_loop+', nc, Atm, Land, Ice)
```
Comment (rem1776) [inline full/coupler_main.F90:450]: isn't this same call made earlier in the loop (line 446)?

Comment (mlee03) [reply full/coupler_main.F90:450]: good catch!

**Review by rem1776 — APPROVED** [review]

**Pull request merged**